### PR TITLE
proposal for HCO implementation of the golden image feature

### PR DESCRIPTION
**Pull request by nunnatsa**
Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

### CODE REVIEW

File: design-proposals/golden-image-delivery-and-update-pipeline.md [hunk 1]
```diff
@@ -317,3 +321,30 @@ spec:
...
 
+## HyperConverged example
+```yaml
+apiVersion: hco.kubevirt.io/v1beta1
+kind: HyperConverged
+metadata:
+  name: kubevirt-hyperconverged
+spec:
+  featureGates:
+    dataImportCron: true
```
Comment (davidvossel) [inline design-proposals/golden-image-delivery-and-update-pipeline.md:332]: based on the previous documentation, should this be `dataImportCron` or `enableCommonDataImportCron`?

File: design-proposals/golden-image-delivery-and-update-pipeline.md [hunk 2]
```diff
@@ -72,15 +72,19 @@ The common templates shipped with SSP need to have their DataVolumeTemplate sect
...
+During installation, the HCO bundle's CSV will pass into the HCO operator (as a local yaml file within the HCO image) locations where disk images are sourced from for the automated disk image delivery/update feature. 
 
-This document doesn't include within it's scope how the HCO should expose this feature to the end user. The general expectation is that we will start with a simple tuning that enables toggling automated delivery/updates on and then consider advanced tunings in future releases. The exact structure of where this tuning belongs on the HCO API is left up to the HCO community.
+The end user can opt-in to automatic delivery/updates of these image sources through a tuning on the HyperConverged CR, by setting the `commonDataImportCronEnabled` feature gate to `true`. From there the HCO is responsible for configuring DataImportCrons on the SSP operator's CR.
```
Comment (davidvossel) [inline design-proposals/golden-image-delivery-and-update-pipeline.md:77]: The DataImportCron name is related to the underlying CRD that facilitates the import of updated data to a PVC.

Are you all sure you want to expose that naming as the HCO tuning? The feature being enabled here is more directly related to importing common base images. Would something like "enableCommonBootImageImport" make more sense?

**Review by davidvossel — COMMENTED** [review]
in general, the approach makes sense to me. I just had a question about the feature gate name.

**Comment by nunnatsa** [comment]
Makes sense. I'll change the name of the FG.

File: design-proposals/golden-image-delivery-and-update-pipeline.md (hunk 2 above)
Comment (nunnatsa) [reply design-proposals/golden-image-delivery-and-update-pipeline.md:77]: Thank @davidvossel, that makes sense. I'll rename the feature gate.

**Review by davidvossel — APPROVED** [review]
/lgtm

**Comment by mazzystr** [comment]
I see we have a lgtm from @davidvossel.  I will now approve for merge.

/approve

**Pull request merged**